mempool/cat/cache.go

```diff
@@ -0,0 +1,291 @@
+package cat
+
+import (
+	"container/list"
+	"time"
+
+	tmsync "github.com/tendermint/tendermint/libs/sync"
+	"github.com/tendermint/tendermint/types"
+)
+
+// LRUTxCache maintains a thread-safe LRU cache of raw transactions. The cache
+// only stores the hash of the raw transaction.
+// NOTE: This has been copied from mempool/cache with the main diffence of using
+// tx keys instead of raw transactions.
+type LRUTxCache struct {
+	staticSize int
+
+	mtx      tmsync.Mutex
+	cacheMap map[types.TxKey]*list.Element
+	list     *list.List
+}
```

Comment:
We can change the existing one. The reason why I avoided that was because it was important that I didn't touch any of the existing implementations at all.